packages/core/src/lib/social.ts

```diff
@@ -41,16 +48,40 @@ export const getUserInfoFromInteractionResult = async (
   connectorId: string,
   interactionResult?: InteractionResults
 ): Promise<SocialUserInfo> => {
-  const result = z
+  const parse = z
     .object({
       socialUserInfo: z.object({
         connectorId: z.string(),
         userInfo: socialUserInfoGuard,
       }),
     })
-    .parse(interactionResult);
+    .safeParse(interactionResult);
 
+  if (!parse.success) {
+    throw new RequestError('session.connector_session_not_found');
+  }
+
+  const result = parse.data;
   assertThat(result.socialUserInfo.connectorId === connectorId, 'session.connector_id_mismatch');
 
   return result.socialUserInfo.userInfo;
 };
+
+export const findSocialRelatedUser = async (
+  info: SocialUserInfo
+): Promise<null | [string, User]> => {
+  // If both phone and email exist, take phone for priority.
```

Comment:
will it be better to put this comment above in a js-doc format then we can know the logic when calling the function?